src/handlers/v1/markets/[chainId]/[marketId]/deleverage/index.ts

```diff
@@ -0,0 +1,160 @@
+  newInternalServerError,
+} from 'src/libs/api';
+import { MarketInfo, Service, calcHealthRate, calcNetAPR, calcUtilization } from 'src/libs/compound-v3';
+import * as apisdk from '@protocolink/api';
+import * as common from '@protocolink/common';
+import { utils } from 'ethers';
+import { validateMarket } from 'src/validations';
+
+type GetDeleverageQuotationRouteParams = EventPathParameters<{ chainId: string; marketId: string }> &
+  EventBody<{ account?: string; token?: common.TokenObject; amount?: string; slippage?: number }>;
+
+type GetDeleverageQuotationResponseBody = QuoteAPIResponseBody<DeleverageQuotation>;
+
+export const v1GetDeleverageQuotationRoute: Route<GetDeleverageQuotationRouteParams> = {
+  method: 'POST',
+  path: '/v1/markets/{chainId}/{marketId}/deleverage',
+  handler: async (event) => {
+    const chainId = Number(event.pathParameters.chainId);
+    const marketId = event.pathParameters.marketId.toUpperCase();
+    if (!validateMarket(chainId, marketId)) {
+      throw newHttpError(400, { code: '400.1', message: 'market does not exist' });
+    }
+
+    if (!event.body) {
+      throw newHttpError(400, { code: '400.2', message: 'body is invalid' });
+    }
+
+    let account = event.body.account;
+    if (!account) {
+      throw newHttpError(400, { code: '400.3', message: `account can't be blank` });
+    }
+    try {
+      account = utils.getAddress(account);
+    } catch {
+      throw newHttpError(400, { code: '400.4', message: 'account is invalid' });
+    }
+
+    const service = new Service(chainId);
+
+    let marketInfo: MarketInfo;
+    try {
+      marketInfo = await service.getMarketInfo(marketId, account);
+    } catch (err) {
+      throw newInternalServerError(err);
+    }
+    const { utilization, healthRate, netAPR, borrowUSD } = marketInfo;
+    const currentPosition = { utilization, healthRate, netAPR, totalDebt: borrowUSD };
+
+    const logics: GetDeleverageQuotationResponseBody['logics'] = [];
+    let approvals: GetDeleverageQuotationResponseBody['approvals'] = [];
+    let targetPosition = currentPosition;
+    if (event.body.token && event.body.amount && Number(event.body.amount) > 0) {
+      const { token, amount, slippage } = event.body;
+      const {
+        baseToken,
+        baseTokenPrice,
+        supplyAPR,
+        supplyUSD,
+        borrowAPR,
+        borrowUSD,
+        collateralUSD,
+        borrowCapacityUSD,
+        liquidationLimit,
+        collaterals,
+      } = marketInfo;
+
+      const deleverageCollateralToken = common.Token.from(token);
+      const deleverageCollateral = collaterals.find(({ asset }) => asset.is(deleverageCollateralToken.unwrapped));
+      if (!deleverageCollateral) {
+        throw newHttpError(400, { code: '400.5', message: 'deleverage token is not collateral' });
+      }
+      const deleverageDebtUSD = new BigNumberJS(amount).times(baseTokenPrice);
+
+      // 1. get the quotation for swaping the base token into amount of deleverage token.
```

Comment:
should be `get the quotation for swaping the deleverage token into amount of the base token`

Reply:
btw, `swapping`.